### DISCUSSION TITLE
Implement get_list_or_404()

### DISCUSSION THREAD

**Discussion by yyyyyyyan** [encode/django-rest-framework#8515] [Ideas & Suggestions] [answered]
Are there any reasons why we don't have [`get_list_or_404()`](https://docs.djangoproject.com/en/dev/topics/http/shortcuts/#get-list-or-404) implemented like `get_object_or_404()`? Since the shortcut function is available since before [Django 2.2](https://docs.djangoproject.com/en/2.2/topics/http/shortcuts/#get-list-or-404), I believe it'd be great to have it covered.

**Comment by lovelydinosaur** [comment, marked as answer]
So our default behaviour is to return an empty list if there are no results, rather than a 404.

I think that behaviour is more intuitive than returning a 404 if there are no results.

(I don't think we expose a `get_object_or_404` function to the public API, so I'm not *exactly* clear what API change you're suggesting. Are you suggesting that we change the behaviour of ListAPIView to 404 instead of returning an empty list?)

**Reply by yyyyyyyan** [reply]
I actually didn't realize `get_object_or_404()` is not exposed in the public API!! Thank you for clarifying.

**Reply by zarinpy** [reply]
404 is for situations where we cannot find an object or entity, not a list, list can be empty
